docs/explanations/cdf-blog.md

```diff
@@ -0,0 +1,210 @@
+
+```json
+{
+  "name": "my-app",
+  "version": "2.2.2",
+  "release": "2023-12-08:12-00-00",
+  "platform_id": "linux",
+  "package": "docker",
+  "description": "This is an event for our application build.",
+  "payload": {
+    "database": "postgres"
+  },
+  "success": true,
+  "event_receiver_id": "01HDS785T0V8KTSTDM9XGT33QQ"
+}
+```
+
+### Event Receiver Groups
+
+Event receiver groups can be thought of as gates that control whether an
+artifact advances through the pipeline. Each group comprises multiple receivers.
+Like receivers, groups can cause the generation of Redpanda events. However,
+they only do this if each receiver has an event with a matching NVRPP
+where `success=true`. Since there may be multiple events of varying successes
+per receiver, only the most recent is considered. This allows you to run many
+tasks in parallel, but only advance your artifact through the pipeline once all
+its tasks have completed successfully.
+
+```json
+{
+  "name": "release-checks",
+  "type": "artifact.release",
+  "version": "3.3.3",
+  "description": "Send an event to release our application if all pipeline tasks have passed.",
+  "enabled": true,
+  "event_receiver_ids": [
+    "01H9GW7FYY4XYE2R930YTFM7FM",
+    "01HDS785T0V8KTSTDM9XGT33QQ"
+  ]
+}
+```
+
+### Watchers
+
+Watchers are applications (typically microservices) that watch the EPR Redpanda topic for messages and then take some action. They use some matching logic, defined in EPR's SDK to determine which messages to process. You can match on NVRPP, `type`, and `success`. We've written about a dozen or so watchers interally that do a variety of things. One of our more popular watchers fires webhooks if matching criteria are met. It is most often used to trigger Jenkins jobs, acting as glue between Jenkins and other systems. Another popular watcher is one that creates Jira tickets when messages are matched. We use this one heavily as part of our security automation to open security issues against various teams when problems are detected.
+
+- Show messages on the bus
+
+## Running EPR in Production
+
+Now that you understand the basics, here's a real world example. We'll be starting with a build of the fabulous `my-app` application. At the end of the build process, the build automation will post a passing receipt to EPR. The build automation generates an NVRPP which will be used by the first and subsequent events for this artifact. Downstream watchers consume the successful build event on the Redpanda topic, triggering a security scan, integration tests, and artifact signing. Watchers for each of those tasks invoke them with the NVRPP used in the build event. These three tasks post more events their corresponding receivers that are all contained inside a release group. Once those three receivers have passing events for the NVRPP, the group triggers a release event, that a downstream watcher catches to release the software.
```

Comment:
Watchers header & downward has paragraphs on same line so harder to give inline feedback.

* "post a passing receipt to EPR" -> "post a passing event to EPR"
* "post more events their corresponding receivers" -> "post more events to their corresponding receivers"

Reply:
Hah, I knew there was somewhere I said receipt in here.